src/widget/wtracktableview.cpp

```diff
@@ -79,6 +79,8 @@ WTrackTableView::WTrackTableView(QWidget * parent,
             this, SLOT(setFocus()));
 
     m_pMenu = new WTrackMenu(this, pConfig, m_pTrackCollectionManager);
+    // Receive signal from menu and emit from here.
```

Comment:
I don't think this comment adds anything that isn't clear already from the code.

Reply:
Okay. This was just to indicate that the signal is relayed from here.
On another note, this signal-based interaction makes the usage of menu clumsy. I will try to make the "load to" functionality independent and property of the menu itself.

Reply:
Yes good idea

Reply:
The loadTrackToPlayer() signal is propagated up as:
WTrackMenu -> WTrackTableView -> Library -> PlayerManager
Using this model, the "Load To" function relies on the client to implement the actual load functionality or relay the loadTrackToPlayer signal to another object able to play the track.
Thus this functionality is still tightly bound to the WTrackTableView and still does not seem to be independent.
Would you like to suggest an alternative measure?

Reply:
@Be-ing What should be done about this?

Reply:
Although it is complicated, I think bubbling up the signal like this does make sense. Otherwise, you would have to pass a pointer to PlayerManager all the way down to WTrackMenu.